### DISCUSSION TITLE
AsyncNotifier doesn't reset the variables in notifier after ref.invalidate

### DISCUSSION THREAD

**Discussion by AminSajadiWork** [rrousselGit/riverpod#3150] [Q&A]
Hello

There is a scenario in which I need to invalidate/refresh my AsyncNotifier to start fresh, but when I invalidate it I face 2 issues:

1. I have a variable inside my AsyncNotifier class that after invalidating that provider still holds on to the data that it had before invalidation, then how should I invalidate a provider to make it refresh everything including variables locally in that class?
2. It still returns AsyncData but changes the isLoading into true, but I thought it should set data null and then start the build function, am I wrong?

I'd appreciate it a lot if you could help me!

**Comment by rrousselGit** [comment]
That's expected. AsyncNotifier preserves the last data where possible. Your UI can still react accordingly
